go/tools/gopackagesdriver/gopackagesdriver_test.go

```diff
@@ -186,7 +167,71 @@ func TestExternalTests(t *testing.T) {
 	}
 }
 
+func runForTest(t *testing.T, args ...string) driverResponse {
+	t.Helper()
+
+	// Remove most environment variables, other than those on an allowlist.
+	//
+	// Bazel sets TEST_* and RUNFILES_* and a bunch of other variables.
+	// If Bazel is invoked when these variables, it assumes (correctly)
+	// that it's being invoked by a test, and it does different things that
+	// we don't want. For example, it randomizes the output directory, which
+	// is extremely expensive here. Out test framework creates an output
+	// directory shared among go_bazel_tests and points to it using .bazelrc.
+	//
+	// This only works if TEST_TMPDIR is not set when invoking bazel.
+	// bazel_testing.BazelCmd normally unsets that, but since gopackagesdriver
+	// invokes bazel directly, we need to unset it here.
+	allowEnv := map[string]bool{
+		"HOME":        true,
+		"PATH":        true,
+		"PWD":         true,
+		"SYSTEMDRIVE": true,
+		"SYSTEMROOT":  true,
+		"TEMP":        true,
+		"TMP":         true,
+		"TZ":          true,
+		"USER":        true,
+	}
+	var oldEnv []string
+	for _, env := range os.Environ() {
+		key, value, ok := strings.Cut(env, "=")
+		if ok && !allowEnv[key] {
```

Comment:
```suggestion
	allowEnv := map[string]struct{}{
		"HOME":        {},
		"PATH":        {},
		"PWD":         {},
		"SYSTEMDRIVE": {},
		"SYSTEMROOT":  {},
		"TEMP":        {},
		"TMP":         {},
		"TZ":          {},
		"USER":        {},
	}
	var oldEnv []string
	for _, env := range os.Environ() {
		key, value, ok := strings.Cut(env, "=")
		if allowed := allowEnv[key]; ok && allowed {
```

Reply:
Done, though I needed to adjust it a bit and didn't commit it directly.